web/client/api/CSW.js

```diff
@@ -303,125 +304,136 @@ const Api = {
                 }
                 resolve(axios.post(parseUrl(url), body, { headers: {
                     'Content-Type': 'application/xml'
-                }}).then(
-                    (response) => {
-                        if (response) {
-                            let json = unmarshaller.unmarshalString(response.data);
-                            if (json && json.name && json.name.localPart === "GetRecordsResponse" && json.value && json.value.searchResults) {
-                                let rawResult = json.value;
-                                let rawRecords = rawResult.searchResults.abstractRecord || rawResult.searchResults.any;
-                                let result = {
-                                    numberOfRecordsMatched: rawResult.searchResults.numberOfRecordsMatched,
-                                    numberOfRecordsReturned: rawResult.searchResults.numberOfRecordsReturned,
-                                    nextRecord: rawResult.searchResults.nextRecord
-                                    // searchStatus: rawResult.searchStatus
-                                };
-                                let records = [];
-                                let _dcRef;
-                                if (rawRecords) {
-                                    for (let i = 0; i < rawRecords.length; i++) {
-                                        let rawRec = rawRecords[i].value;
-                                        let obj = {
-                                            dateStamp: rawRec.dateStamp && rawRec.dateStamp.date,
-                                            fileIdentifier: rawRec.fileIdentifier && rawRec.fileIdentifier.characterString && rawRec.fileIdentifier.characterString.value,
-                                            identificationInfo: rawRec.abstractMDIdentification && rawRec.abstractMDIdentification.value
-                                        };
-                                        if (rawRec.boundingBox) {
-                                            let bbox;
-                                            let crs;
-                                            let el;
-                                            if (Array.isArray(rawRec.boundingBox)) {
-                                                el = head(rawRec.boundingBox);
-                                            } else {
-                                                el = rawRec.boundingBox;
-                                            }
-                                            if (el && el.value) {
-                                                const crsValue = el.value?.crs ?? '';
-                                                const urn = crsValue.match(/[\w-]*:[\w-]*:[\w-]*:[\w-]*:[\w-]*:[^:]*:(([\w-]+\s[\w-]+)|[\w-]*)/)?.[0];
-                                                const epsg = makeNumericEPSG(crsValue.match(/EPSG:[0-9]+/)?.[0]);
+                }}).then(async(response) => {
```

Comment:
can't you make it without using async await, it's super hard to compare changes this way

Reply:
I don't think so 
I just put it to enable me to get bbox from getCapabilities before pushing the obj into reords array.

![image](https://github.com/geosolutions-it/MapStore2/assets/58145645/c3220585-a000-4616-b1f8-12f3e3e607c5)

That's only what I added but because of lint rules it made a lot of changes here. 
But let me check if I can do another thing.

Reply:
yeah, but i was wondering if you can chain it with something similar done with addCapabilitiesToRecords function.
what about this ?